In [4]:
'''Blood glucose levels for obese patients have a mean of 100 with a standard deviation of
15. A researcher thinks that a diet high in raw cornstarch will have a positive effect on 
blood glucose levels. A sample of 36 patients who have tried the raw cornstarch diet 
have a mean glucose level of 108. Test the hypothesis that the raw cornstarch had an 
effect or not.'''

p_mean = 100
p_std = 15
n = 36
s_mean = 108

import numpy as np
import math as math
import scipy.stats as stats

z_score = (s_mean - p_mean)/(p_std/math.sqrt(n))
print(f"computed value of z_score is : {z_score}")

# as no significance level is given , we assume it as 0.05 
alpha = 0.05

print(f'critical region is from {stats.norm.ppf(alpha/2)} to {-stats.norm.ppf(alpha/2)}')

computed value of z_score is : 3.2
critical region is from -1.9599639845400545 to 1.9599639845400545


so the value of computed z score is out of critical region, so we reject the null hypothesis, hence the cornstarch has effect

In [2]:
'''Problem Statement 2:
In one state, 52% of the voters are Republicans, and 48% are Democrats. In a second 
state, 47% of the voters are Republicans, and 53% are Democrats. Suppose a simple 
random sample of 100 voters are surveyed from each state.
What is the probability that the survey will show a greater percentage of Republican 
voters in the second state than in the first state? '''

#proportion of republicans in 1 state :
p1 = .52

#proportion of republicans in 2nd state :
p2 = .47

# let n1 & n2 be the sample size for both states , so
n1 = 100
n2 = 100

# diffrence in proportion of republicans in both states 
d = p1 - p2
print(f'the diffrence in proportion of republicans is : {d}')
print()
# standard deviation of the proportion of republicans in both states 
import math as math

s = math.sqrt(( p1*(1-p1)/n1 ) + ( p2*(1-p2)/n2 ))
print(f'the standard deviation of republican is : {s}')

# if we assume that in survey, republicans of 2nd state have more percentage than that of 1st state, then 
# diffrence in their proportions must be less than zero which means that value of diffrence corrosponds to zero means p = 0
p= 0
z = (p - d)/s

print(f' the value of z is : {z}')


the diffrence in proportion of republicans is : 0.050000000000000044

the standard deviation of republican is : 0.07061869440877536
 the value of z is : -0.7080278164104213


by seeing the standard normal distribution z table , the value of z is -0.70 corrosponds to p value 0.239
so we can say that the probablity that survey will show greater percentage of republican is 0.24 ie. 24%

In [1]:
'''Problem Statement 3:
You take the SAT and score 1100. The mean score for the SAT is 1026 and the standard deviation is 209.
How well did you score on the test compared to the average test taker?'''

p_mean = 1026
p_std = 209
s_mean = 1100

z = (s_mean - p_mean)/p_std

print(f'our z score is : {z}')


our z score is : 0.35406698564593303


as our z value is 0.35, it means we have performed .35 % more than the average test taker